doc/user/csv-annotations.md

```diff
@@ -0,0 +1,147 @@
+# Code Annotations for Cluster Service Versions
+
+## Overview
+
+This document describes the semantics of Cluster Service Version (CSV) [code annotations][code_annotations_design] and lists all possible annotations.
+
+## Usage
+
+All annotations have a `+operator-sdk:gen-csv:` prefix, denoting that they're parsed while executing [`operator-sdk olm-catalog gen-csv`][sdk_cli_ref].
```

Comment:
Thinking ahead a little bit, do you think the `+operator-sdk:<cli-command>` pattern will cause us problems when we start implementing the CLI UX improvements?  

I could envision ending up with any of the following:
- `operator-sdk generate csv` (moved under global generate subcommand)
- `operator-sdk olm gen-csv` (rename `olm-catalog` to `olm` to accomodate non-catalog OLM commands)
- `operator-sdk olm generate csv` (new OLM subcommand that mirrors non-OLM commands)

Not sure what the right answer is here. Maybe we relax the pattern to be less coupled to the CLI-command names? If we stick with a command-based pattern and end up changing the CLI command name, would we have to deprecate the old annotation name and add a new annotation name for the new command name?

Reply:
Yeah we would have to change annotation prefixes because the command name or parent command name no longer exist. I agree we should relax this requirement. Instead we should couple the prefix to how the annotation is used. Keeping `gen-csv` here is fine imo because this annotation class is used in CSV generation.